### ISSUE TITLE
Enable navigation between documentation versions

### ISSUE THREAD

**Issue by sef43** [openmm/openmm#3992]
It is not currently possible to navigate between OpenMM documentation versions, you have to directly edit the url, e.g.
7.7: http://docs.openmm.org/7.7.0/userguide/
latest: http://docs.openmm.org/latest/userguide/
development: http://docs.openmm.org/development/userguide/
It would be useful to have a way of navigating between them, e.g. this functionality from [MDAnalysis](https://docs.mdanalysis.org/stable/index.html) 
<img width="200" alt="Screenshot 2023-03-08 at 14 36 10" src="https://user-images.githubusercontent.com/25941757/223742477-aef84ecf-9db9-4a7c-b913-c33e344cf7f8.png">

Additionally, to a user arriving from google (#3991 ) it is not immediately clear which version of the Documentation they are on. It would be useful to have a message telling them they are on an old version, e.g. like on Numpy Docs: https://numpy.org/doc/1.17/user/index.html

**Comment by peastman** [comment]
Sounds like a good idea.

**Comment by aizvorski** [comment]
Second this :100: 

Search engines don't seem to have the current version of the documentation as the top result, if they return it at all.  For example google "openmm HarmonicBondForce" - the top results are for version 7.0.0, 7.4.0, 3.0 and 6.1 all with version numbers in the urls; docs/latest doesn't come up.  So it's easy to end up on an old version.

**Comment by peastman** [comment]
Is there some way we can tell search engines to only index the `latest` version, not the ones for specific older versions?

**Comment by bdenhollander** [comment]
Search engines indexing behaviour can be requested using robots.txt. I would advise against it since users may only have access to a specific version on their cluster so they need that version of the documentation.

The version of the latest documentation is not obvious i.e. no version number, release date, recent changes, new features, deprecated features, etc. Improving that would also be helpful, even if it's just a link to the release announcement on GitHub.

**Comment by sef43** [comment]
I have a simple of concept of this here: https://sef43.github.io/openmm_docs/api-python/
The top of the left side bar says the version number, with a link to the specific openmm git commit it was built with.
The drop down box links to the older versions I could easily find:

<img width="256" alt="Screenshot 2023-04-13 at 17 00 44" src="https://user-images.githubusercontent.com/25941757/231818228-946e8ea1-44fc-4c84-a9e7-359f704b2cbb.png">


what are peoples thoughts on this? @peastman 

Is there a list of the domains of the older docs versions?

**Comment by peastman** [comment]
Here's the list of folders in the docs bucket.

6.1.0
6.2.0
6.3.0
7.0.0
7.1.0
7.2.0
7.3.0
7.4.0
7.5.0
7.6.0
7.7.0
8.0.0
development
latest

Putting it in the sidebar like that means it's up to each docs folder to provide it.  If you select an older version, it disappears.  And presumably that would mean older versions would never list newer ones?  For example, we release docs for 8.1, and they list 8.1 as the latest version.  Then we release 8.2, but when someone views the 8.1 docs they don't list 8.2 as an option.

Is there a way we could either put it in an outer frame, or inject it into the sidebar for older versions at a correct place?

**Comment by sef43** [comment]
yes we can just have a separate hosted versions.json file that could just live somewhere in the docs bucket, (would need to be manually updated for a new release), e.g. like this here: https://sef43.github.io/openmm_docs/versions.json
The 8.1 release docs would read this file using JavaScript, so when 8.2 is released the 8.1 will have a link to 8.2
(This is how the  [pydata-sphinx-theme](https://pydata-sphinx-theme.readthedocs.io/en/stable/index.html) does it which numpy docs use)

**Comment by peastman** [comment]
That still leaves every version up to now not displaying it.  It doesn't achieve one of the main goals: to tell the user when they've been taken to an old version that got indexed by a search engine, and to let them easily switch to the current version.

**Comment by sef43** [comment]
Yes that is true. I think that will be more involved. This is out of my area of expertise. I guess to achieve that we will need to edit all of the older docs that are currently hosted. Either by programatically adding some HTML banner to the header, or rebuilding the older docs from checking out the specific git release tags, and using a modified html template with the version info and switcher.

**Comment by peastman** [comment]
Related to this, we should probably add a `robots.txt` file to `docs.openmm.org` to tell search engines they should only index the `latest` folder.  I believe the following would do it:

```
User-agent: *
Disallow: /
Allow: /latest/
```

Does that look right?  Should I add it?
